### PR TITLE
setup markdownlint

### PR DESCRIPTION
**Pull request by HarshCasper**
## Introduction

This PR introduces configuration files for `markdownlint` [1] and the related `markdownlint-cli2` [2]. Additionally, we have introduced the custom Markdownlint rule `max-one-sentence-per-line` [3] to ensure that we adher to one-sentence-per-line rule going forward.

## Ignored Rules

As part of this exercise, we opted to ignore a few rules, in the interest of time and effort. They are:

-   [**MD029**](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md029): Ordered list item prefix
-   [**MD046**](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md046): Code block style
-   [**MD025**](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md025): Single H1
-   [**MD001**](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md001): Header levels increment by one
-   [**MD024**](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md024): Multiple headers with the same content
-   [**MD055**](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md055): Inconsistent leading and trailing pipe characters
-   [**MD056**](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md056): Inconsistent Table column count
-   [**MD036**](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md036): Emphasis used instead of a header
-   [**MD003**](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md003): Header style
-   [**MD033**](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md033): Inline HTML
-   [**MD013**](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md013): Line length
-   [**MD034**](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md034): Bare URL used
-   [**MD032**](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md032): Lists should be surrounded by blank lines

We hope to progressively fix them over the time but they are not the current priority. 

## GitHub Action

This PR also introduces a GitHub Action [4] that runs **only on changed files** every pull request.

## Pre-Commit Hook

This PR also introduces a Pre-commit hook to run the lint process locally to reduce the feedback cycle.

## Next Steps 

Service owners should review the docs that they own and report to us if Markdownlint has inadvertently made unexpected changes. 

[1]: https://github.com/DavidAnson/markdownlint
[2]: https://github.com/DavidAnson/markdownlint-cli2
[3]: https://github.com/aepfli/markdownlint-rule-max-one-sentence-per-line
[4]: https://github.com/DavidAnson/markdownlint-cli2-action

### CODE REVIEW

**Review by alexrashed — COMMENTED** [review]
It would be great if you could add a description to the PR such that reviewers can get an insight of what you want to accomplish with the changes here.

**Comment by HarshCasper** [comment]
@alexrashed — I was in the process of writing that :) 

I hope the current description gives an idea of what we are looking to accomplish.

**Review by alexrashed — COMMENTED** [review]
Awesome! Thanks for the detailed explanation! Automation / linting around our best practices has often proven to be very effective and helpful!

In other repositories we use [pre-commit](https://pre-commit.com/) ([there is an offical one for markdownlint](https://github.com/markdownlint/markdownlint/blob/main/.pre-commit-hooks.yaml)), which helps contributors to get notified on issues in their files right when they commit the changes locally.
It would also be great if you could fix the last remaining issues in the code, if there's a lot we could also try to get codeowners on board?

And another tip on the side (which I think won't be necessary for this changeset): If you want to protect the Git history / blame views from being polluted by formatter changes, you can use [`.git-blame-ignore-revs`](https://github.com/localstack/localstack/blob/master/.git-blame-ignore-revs) to ignore specific commits (later on). However, for that you would have to separate the configuration of the linting tools from the actual formatting...

**Comment by HarshCasper** [comment]
Thank You @alexrashed for the feedback — I'll setup the pre-commit hook and revisit the [.git-blame-ignore-revs](https://github.com/localstack/localstack/blob/master/.git-blame-ignore-revs) later on 👍

File: content/en/references/coverage/coverage_account/index.md
Comment (steffyP) [inline content/en/references/coverage/coverage_account/index.md:1]: @HarshCasper: just as an FYI - all the coverage files are auto-generated (by the `Update Parity Docs` workflow) - I think we should either make sure that the workflow runs the format as well, or adapt the template for the file creation :)

Comment (HarshCasper) [reply content/en/references/coverage/coverage_account/index.md:1]: I will adapt the workflow to run the format as well 👍

File: content/en/user-guide/aws/ec2/index.md
Comment (viren-nadkarni) [inline content/en/user-guide/aws/ec2/index.md:321]:
```suggestion
#!/bin/bash
```

File: content/en/references/cross-account-access.md
Comment (viren-nadkarni) [inline content/en/references/cross-account-access.md:22]:
```suggestion
```

File: content/en/references/cross-account-access.md
Comment (viren-nadkarni) [inline content/en/references/cross-account-access.md:24]:
```suggestion
```

File: content/en/references/cross-account-access.md
Comment (viren-nadkarni) [inline content/en/references/cross-account-access.md:35]:
```suggestion
```

File: content/en/references/cross-account-access.md
Comment (viren-nadkarni) [inline content/en/references/cross-account-access.md:37]:
```suggestion
```

File: content/en/references/cross-account-access.md
Comment (viren-nadkarni) [inline content/en/references/cross-account-access.md:50]:
```suggestion
# Retrieve the queue attribute from another account and region
# The required information for LocalStack to locate the queue is available in the queue URL
```

**Review by remotesynth — APPROVED** [review]
I didn't go through all 336 changed files of course, but the preview looks fine and the additions seem fine as well.

**Review by macnev2013 — APPROVED** [review]
Thanks for jumping on it. I've gone through my services. LGTM 🚀

**Review by viren-nadkarni — APPROVED** [review]
I recommend verifying all changes yourself because not all pages have owners.

**Review by steffyP — APPROVED** [review]
Nice work! I think having this formatting in place will make things moving forward. 🚀 

My only remarks left:
1. please add a note to the README regarding the pre-commit hook
2. please make sure that the workflow for updating parity docs still work - either trigger the workflow (against this branch), or after we merged this. Only want to make sure everything runs smoothly 🙂

**Pull request merged**
